### ISSUE TITLE
Disable caching when using a reverse proxy

### ISSUE THREAD

**Issue by ProphetLamb** [digitalocean/nginxconfig.io#467]
Generate a `globals.conf` without media caching for reverse proxy consumption

## Information

The routing does not pass through the proxy when these settings are present in the `global.conf`, instead nginx attempts to route these requests via the default site.
The directory of the default site doesn't contain the `css` or `js` resources provides via the for example node runtime the proxy pass delegates to.

### Description

This is caused by the location handling in `general.conf` included in every reverse proxy domain.

```conf
# assets, media
location ~* \.(?:css(\.map)?|js(\.map)?|jpe?g|png|gif|ico|cur|heic|webp|tiff?|mp3|m4a|aac|ogg|midi?|wav|mp4|mov|webm|mpe?g|avi|ogv|flv|wmv)$ {
    expires 7d;
}

# svg, fonts
location ~* \.(?:svgz?|ttf|ttc|otf|eot|woff2?)$ {
    add_header Access-Control-Allow-Origin "*";
    expires    7d;
}
```

### Steps to reproduce

1. Create a basic node application and serve at `localhost:3000`.
2. Create a [NGINX Config](https://www.digitalocean.com/community/tools/nginx?domains.0.server.domain=app.example.com&domains.0.php.php=false&domains.0.reverseProxy.reverseProxy=true&domains.0.routing.root=false&domains.0.routing.index=index.html&domains.0.routing.fallbackHtml=true&global.app.lang=de) with a proxy pass from the subdomain `app.example.com` to `localhost:3000`.
3. Open your browser devtools network tab
4. Request `app.example.com` in your browser
5. media types handled in `general.conf` will fail with 404.

### Expected behavior

Resources are requested from the reverse proxy

### Screenshots

![example](https://github.com/digitalocean/nginxconfig.io/assets/19748542/4cf0eeb8-c81b-4f7e-bb10-5063ab02a385)

```log
-  - [25/Feb/2024:19:51:08 +0000] "GET /_app/immutable/chunks/entry.aDhfE7Nt.js HTTP/2.0" 404 548 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36"
-  - [25/Feb/2024:19:51:08 +0000] "GET /_app/immutable/chunks/index.CA4DBg1v.js HTTP/2.0" 404 548 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36"
```

**Comment by NinZeige** [comment]
Yes, this issue cost me several hours today. While trying to use the generated configuration file to reverse proxy a Gitea service on the host, I found that many resource files (like `index.js`) were not loading properly.

In a natural setup, which I believe, we would certainly want the forwarded service to independently decide which to return. However, due to the generated files including `nginxconfig.io/general.conf`, the forwarding is blocked, which makes the auto-generated reverse proxy configuration almost *unusable* 😕 (I hate to manually modify an auto-generated file).

A simple workaround would be to avoid inserting the content of `general.conf` into the reverse proxy configuration file. Upon further thought, one also agree that the logic in that file should actually be handled by the proxied service itself, not by Nginx.

**Comment by mellowcandle** [comment]
I wasted 3 hours debugging it. I agree with everything that was written. this needs to be fixed.
